web-marketplace/src/lib/ecocredit/api.ts

```diff
@@ -421,7 +423,19 @@ export const queryProjectsByClass = async ({
   request,
 }: QueryProjectsByClassProps): Promise<QueryProjectsByClassResponse> => {
   try {
-    return client.ProjectsByClass({ classId: request.classId });
+    let projects: ProjectInfo[] = [];
+    let response: QueryProjectsByClassResponse | undefined;
+    while (!response || !!response.pagination?.nextKey) {
```

Comment:
I think the issue with the duplicate projects is that `!!response.pagination?.nextKey` evaluates to true because after the first loop, `response.pagination?.nextKey` is defined, although it's an empty `Uint8Array` so we should probably check for its length instead
ditto in all queries

Reply:
Good catch! This seems to fix it. I wonder why this didn't cause some kind of an infinite loop?

Reply:
sorry as discussed during stand up, I got confused with something else that triggered infinite re-rendering

Reply:
Done: https://github.com/regen-network/regen-web/pull/2356/commits/23736684b04b9787a2e60bf38e71ed2b7f09de4d